shared/util/arrays.js

```diff
@@ -6,13 +6,13 @@ export function intersperseFn<A, B>(
   arr: Array<A>
 ): Array<A | B> {
   if (arr.length === 0) {
-    return []
+    // $FlowIssue an array with no length is a valid type for any Array
+    return arr
   }
```

Comment:
Could try and make a local module const that's returned (this is fine also. Making a new array each time likely bad)